src/ovs/aca_ovs_control.cpp

```diff
@@ -0,0 +1,326 @@
+// #include <unistd.h>
+#include <pcap.h>
+#include <net/ethernet.h>
+#include <netinet/ether.h>
+#include <netinet/ip.h>
+#include <arpa/inet.h>
+
+using namespace std;
+using namespace ovs_control;
+
+extern string g_ofctl_command;
+extern string g_ofctl_target;
+extern string g_ofctl_options;
+
+namespace aca_ovs_control
+{
+ACA_OVS_Control &ACA_OVS_Control::get_instance()
+{
+  // Instance is destroyed when program exits.
+  // It is instantiated on first use.
+  static ACA_OVS_Control instance;  
+
+  return instance;
+}
+
+int ACA_OVS_Control::control()
+{
+    int overall_rc = EXIT_SUCCESS;
+
+    char target[g_ofctl_target.size() + 1];
+    g_ofctl_target.copy(target, g_ofctl_target.size() + 1);
+    target[g_ofctl_target.size()] = '\0';
+
+    char options[g_ofctl_options.size() + 1];
+    g_ofctl_options.copy(options, g_ofctl_options.size() + 1);
+    options[g_ofctl_options.size()] = '\0';
+
+    if (g_ofctl_command.compare("monitor") == 0) {
+        monitor(target, options);
+    } else if (g_ofctl_command.compare("dump-flows") == 0) {
+        dump_flows(target, options);
+    } else if (g_ofctl_command.compare("packet-out") == 0) {
+        packet_out(target, options);
+    } else {
+        cout << "Usage: -c <command> -t <target> -o <options>" << endl;
+        cout << "   commands: monitor, dump-flows, packet-out..." << endl;
+        cout << "   target: swtich name, such as br-int, br-tun, ..." << endl;
+        cout << "   options: " << endl;
+        cout << "      moinor: \"[miss-len] [invalid-ttl] [resume] [watch:format]\"" << endl;
+        cout << "      packet-out: \"in_port=<in_port> packet=<hex string> [actions=<actions>]\"" << endl;
+    }
+
+    return overall_rc;
+}
+
+int ACA_OVS_Control::dump_flows(const char *bridge, const char *opt) 
+{
+    int overall_rc = EXIT_SUCCESS;
+    OVS_Control::get_instance().dump_flows(bridge, opt);
+    return overall_rc;
+}
+
+int ACA_OVS_Control::monitor(const char *bridge, const char *opt) 
+{
+    int overall_rc = EXIT_SUCCESS;
+    OVS_Control::get_instance().monitor(bridge, opt);
+    return overall_rc;
+}
+
+int ACA_OVS_Control::packet_out(const char *bridge, const char *opt) 
+{
+    int overall_rc = EXIT_SUCCESS;
+    OVS_Control::get_instance().packet_out(bridge, opt);
+    return overall_rc;
+}
+
+void ACA_OVS_Control::parse_packet(void *packet)
+{
+    const struct ether_header *eth_header;
+    /* The packet is larger than the ether_header struct,
+    but we just want to look at the first part of the packet
+    that contains the header. We force the compiler
+    to treat the pointer to the packet as just a pointer
+    to the ether_header. The data payload of the packet comes
+    after the headers. Different packet types have different header
+    lengths though, but the ethernet header is always the same (14 bytes) */
+    eth_header = (struct ether_header *) packet;
+    
+    printf("Destination Mac: %s\n", ether_ntoa((ether_addr *) &eth_header->ether_dhost));
```

Comment:
need to convert the printf into ACA_LOG*

Reply:
... if that can be done quickly.

Reply:
Those printf has been changed to ACA_LOG_INFO().